functions/go/set-setter/setters_transformer_test.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func runSetterTransformerE(config, input string) (string, error) {
+	resmapFactory := newResMapFactory()
+	resMap, err := resmapFactory.NewResMapFromBytes([]byte(input))
+	if err != nil {
+		return "", err
+	}
+
+	var plugin = KustomizePlugin
+	err = plugin.Config(nil, []byte(config))
+	if err != nil {
+		return "", err
+	}
+	err = plugin.Transform(resMap)
+	if err != nil {
+		return "", err
+	}
+	y, err := resMap.AsYaml()
+	if err != nil {
+		return "", err
+	}
+	return string(y), nil
+}
+
+func TestSetCommand(t *testing.T) {
+	var tests = []struct {
+		name              string
+		config            string
+		input             string
+		expectedResources string
+		errMsg            string
+	}{
+		{
+			name: "set name and label",
+			input: `apiVersion: v1
+kind: Service
+metadata:
+  name: myService # kpt-set: ${app}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: mungebot # kpt-set: ${app}
+  name: mungebot
+`,
+			config: `
+setters:
```

Comment:
is setting subpckages a future change? If so, make it explicit it in the PR description and add TODOs.

Are there tests for non-scalar values?

Reply:
Yes. Updated the description.